packages/ui/src/components/TimeInputV2/index.tsx

```diff
@@ -0,0 +1,469 @@
+        .toString()
+        .padStart(2, '0')
+    } else {
+      newTime[type] = (currentValue === 59 ? 0 : currentValue + 1)
+        .toString()
+        .padStart(2, '0')
+    }
+    setTime(newTime)
+    onChange?.(newTime)
+  }
+
+  // Decrease time with arrow down
+  const handleDecrease = (type: 'h' | 'm' | 's') => {
+    const newTime = { ...time }
+    const currentValue = Number.parseInt(
+      time[type] && time[type].length > 0 ? time[type] : '0',
+      10,
+    )
+
+    if (type === 'h' && timeFormat === 24) {
+      newTime[type] = (currentValue === 0 ? 23 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    } else if (type === 'h' && timeFormat === 12) {
+      newTime[type] = (currentValue === 1 ? 12 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    } else {
+      newTime[type] = (currentValue === 0 ? 59 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    }
+    setTime(newTime)
+    onChange?.(newTime)
+  }
+
+  // Go to next input
+  const handleNext = (type: 'h' | 'm' | 's') => {
+    if (type === 'h') refMinutes.current?.focus()
+    if (type === 'm') refSeconds.current?.focus()
+    if (type === 's' && timeFormat === 12) refPeriod.current?.focus()
+  }
+
+  // Go to previous input
+  const handlePrevious = (type: 'h' | 'm' | 's') => {
+    if (type === 'm') refHours.current?.focus()
+    if (type === 's') refMinutes.current?.focus()
+  }
+
+  return (
+    <Stack gap={0.5} className={className} id={id} data-testid={dataTestId}>
+      <Stack direction="row" gap={1} alignItems="center">
+        <Text as="label" prominence="strong" sentiment="neutral" variant="body">
+          {label}
+        </Text>
+        {required ? <AsteriskIcon size={8} sentiment="danger" /> : null}
+        {labelDescription ? (
+          <Text as="label" variant="bodySmall">
+            {labelDescription}
+          </Text>
+        ) : null}
+        {labelDescription && typeof labelDescription !== 'string'
+          ? labelDescription
+          : null}
+      </Stack>
+      <TimeInputWrapper
+        data-readonly={readOnly}
+        data-disabled={disabled}
+        data-size={size}
+        data-error={!!error}
+        direction="row"
+        alignItems="center"
+        justifyContent="space-between"
+        onBlur={onBlur}
+        onFocus={onFocus}
+        aria-required={required}
+      >
+        <Stack direction="row">
+          {timeKeys.map(type => {
+            const computedRef = () => {
+              if (type === 'h') return refHours
+              if (type === 'm') return refMinutes
+
+              return refSeconds
+            }
+            const fullName = () => {
+              if (type === 'h') return 'hours'
+              if (type === 'm') return 'minutes'
+
+              return 'seconds'
+            }
+
+            return (
+              <Stack key={type} gap={0} direction="row">
```

Comment:
```suggestion
              <Stack key={type} direction="row">
```

I don't think its needed